src/components/Calculator.js

```diff
@@ -1,123 +1,82 @@
 
-  componentDidMount() {
-    const expression = expressionStorage.getExpression();
-    if (!expression) {
-      return;
-    }
-    const { sum, prevNumber, operator, nextNumber } = expression;
-    this.setState({ sum, prevNumber, operator, nextNumber });
-  }
+const Calculator = () => {
+  const [state, setState] = useState(CALCULATOR_INITIAL_STATE);
+  const stateRef = useRef(state);
 
-  confirmExist = (e) => {
-    e.preventDefault();
-    e.returnValue = CONFIRM_MSG;
-    const { sum, prevNumber, operator, nextNumber } = this.state;
-    expressionStorage.setExpression({ sum, prevNumber, operator, nextNumber });
-  };
+  useEffect(() => {
+    stateRef.current = state;
+  });
 
-  onClickNumber = (e) => {
-    const number = e.target.dataset.number;
-    const isPrev = this.state.operator === '';
-    try {
-      if (isPrev) {
-        if (this.state.prevNumber.length >= NUMBER_LIMIT) {
-          throw new Error(ERROR_MSG.OVER_NUMBER_LIMIT);
-        }
-        this.setState({ prevNumber: [...this.state.prevNumber, number] });
-        return;
-      }
-      if (this.state.nextNumber.length >= NUMBER_LIMIT) {
-        throw new Error(ERROR_MSG.OVER_NUMBER_LIMIT);
-      }
-      this.setState({ nextNumber: [...this.state.nextNumber, number] });
-    } catch ({ message }) {
-      alert(message);
-    }
-  };
+  useEffect(() => {
+    setInitialState();
+    window.addEventListener('beforeunload', confirmExist);
+    return () => {
+      window.removeEventListener('beforeunload', confirmExist);
+    };
+  }, []);
 
-  onClickOperator = (e) => {
-    const operand = e.target.dataset.operator;
-    if (operand === OPERATOR.EQUAL) {
-      const prevNumbers = Number(this.state.prevNumber.join(''));
-      const nextNumbers = Number(this.state.nextNumber.join(''));
-      switch (this.state.operator) {
-        case OPERATOR.PLUS:
-          this.setState({ sum: prevNumbers + nextNumbers });
-          break;
-        case OPERATOR.SUBSTRACT:
-          this.setState({ sum: prevNumbers - nextNumbers });
-          break;
-        case OPERATOR.MULTI:
-          this.setState({ sum: prevNumbers * nextNumbers });
-          break;
-        case OPERATOR.DIVIDE:
-          if (!isFinite(prevNumbers / nextNumbers)) {
-            this.setState({ sum: ERROR_MSG.INFINITY });
-            break;
-          }
-          this.setState({ sum: prevNumbers / nextNumbers });
-        default:
-          return;
-      }
-    }
-    this.setState({ operator: operand });
-  };
+  const setInitialState = () => {
+    const expression = expressionStorage.getExpression();
+    if (!expression) return;
 
-  onClickAllClear = () => {
-    this.setState({
-      sum: '',
-      prevNumber: [],
-      operator: '',
-      nextNumber: [],
-    });
+    const { sum, prevNumbers, operator, nextNumbers } = expression;
+    setState(prevState => ({
+      ...prevState,
+      sum,
+      prevNumbers,
+      operator,
+      nextNumbers,
+    }));
```

Comment:
결국 state 를 모두 업데이트하고 있는 것 같은데요, prevState 를 사용하는 이유가 있나요?

```suggestion
    setState({ sum, prevNumbers, operator, nextNumbers });
```

Reply:
이전에 local storage에 저장이 안돼서 이것저것 다 해보다가 넘어갔었고 수정했습니다.